crates/goose/src/providers/groq.rs

```diff
@@ -0,0 +1,74 @@
+use crate::message::Message;
+use crate::providers::base::{Provider, ProviderUsage, Usage};
+use crate::providers::configs::{GroqProviderConfig, ModelConfig, ProviderModelConfig};
+use crate::providers::openai_utils::{
+    create_openai_request_payload, get_openai_usage, openai_response_to_message,
+};
+use crate::providers::utils::{get_model, handle_response};
+use async_trait::async_trait;
+use mcp_core::Tool;
+use reqwest::Client;
+use serde_json::Value;
+use std::time::Duration;
+
+pub const GROQ_API_HOST: &str = "https://api.groq.com";
+pub const GROQ_DEFAULT_MODEL: &str = "llama-3.3-70b-versatile";
+
+pub struct GroqProvider {
+    client: Client,
+    config: GroqProviderConfig,
+}
+
+impl GroqProvider {
+    pub fn new(config: GroqProviderConfig) -> anyhow::Result<Self> {
+        let client = Client::builder()
+            .timeout(Duration::from_secs(600)) // 10 minutes timeout
+            .build()?;
+
+        Ok(Self { client, config })
+    }
+
+    fn get_usage(data: &Value) -> anyhow::Result<Usage> {
```

Comment:
we should consider adding get_usage as a trait for Providers in the base object

Reply:
agree. This PR has many changes. I can create a follow-up PR after this PR is merged